README.rst

```diff
@@ -122,6 +122,66 @@ For example, a custom drill-down that offers only past dates:
                     ) if day <= today
                 )
 
+
+EnhancedDateHierarchy
+----------------------------------------------
+
+Django filter the query set for a given level in the date hierarchy using a database
+function to extract the relevent date part. For example, when filtering a queryset on
+a `created` date field for November 2017 Django will execute the following query:
+
+.. code-block:: sql
+
+    SELECT
+        ...
+    FROM
+        app_table
+    WHERE
+        created BETWEEN '2017-01-01 00:00:00' AND '2017-12-31 23:59:59.999999'
+        AND EXTRACT('month', created) = 11
+
+Using a function to filter the queryset often leads to poor performance. The database
+optimizer cannot always estimate the cardinality of such condition which might lead to
+a sub optimal execution plan.
+
+There are several approaches to tackle this issue. For exmaple, in databases that support
+function based indexes (FBI) the developer can add an index on the specific function to
+try and improve the performace of the query. The downside to this approach is having to
+maintain additional indexes for each level of the hierarchy. Additional indexes slow
+insert and update operations and take up space.
+
+Another approach is to simplify the condition used by Django to filter the queryset
+for any given level in the hierarchy:
+
+.. code-block:: sql
+
+    SELECT
+        ...
+    FROM
+        app_table
+    WHERE
+        created >= '2017-11-01 00:00:00'
+        AND created < '2017-12-01 00:00:00'
+
+
+To acheive the above query, simply add the following to your ModelAdmin:
```

Comment:
I would write it as "This is what `EnhancedDateHierarchyListFilter` does; to achieve the above query ...".